### DISCUSSION TITLE
Is it possible to export the same result twice under different names?

### DISCUSSION THREAD

**Discussion by f11r** [pulumi/pulumi#9408] [Pulumi in Practice] [answered]
For compatibility reasons I'm trying to export the same result value twice under different names. However, this does not seem to be possible as only one of the exports is actually created. Is this not possible? The following shows a minimal reproducible example of what I'm trying to do (not actually trying to export a password twice but it produces the same issue). My expectation would be that both `export1` and `export2` get exported, while only the first one does. Is this something that's expected? If yes, is there some way work around this?

```
import pulumi
import pulumi_random

a = pulumi_random.RandomPassword("A", length=32)

pulumi.export("export1", a.result)
pulumi.export("export2", a.result)
```

**Comment by cnunciato** [comment, marked as answer]
Good question! This feels to me like it might be a bug. This works:

```python
...
b = "some-string"
pulumi.export("export3", b)
pulumi.export("export4", b)
```

```bash
...
Outputs:
    export1: [secret]
  + export3: "some-string"
  + export4: "some-string"
```

As does running both outputs though `apply`:

```python
...
pulumi.export("export1", a.result.apply(lambda result: result))
pulumi.export("export2", a.result.apply(lambda result: result))
```

```bash
...
Outputs:
    export1: [secret]
  + export2: [secret]
```

The equivalent program in TypeScript seems to work as well:

```typescript
import * as random from "@pulumi/random";

const a = new random.RandomPassword("A", {
    length: 32,
});

export const export1 = a.result;
export const export2 = a.result;
```

```bash
...
Outputs:
    export1: [secret]
    export2: [secret]
```

So I'm thinking unless this is probably worth filing a bug, as it does seem unexpected and inconsistent with other languages. I'm happy do to that (unless you'd like to).

Let me know if the workaround above (with `.apply(lambda result: result)`) works for you, though. 🤞

**Reply by f11r** [reply]
The `apply` workaround does indeed help. Thanks :+1: 

I've also created a bug report in https://github.com/pulumi/pulumi/issues/9411

**Reply by cnunciato** [reply]
Awesome, thanks for that!
